### DISCUSSION TITLE
I use google voice

### DISCUSSION THREAD

**Discussion by CaoHoa1** [chrisron95/home-assistant-config#5] [Q&A]
I use google voice cast media player and get the following error

**Comment by CaoHoa1** [comment]
![image](https://user-images.githubusercontent.com/31616290/226825543-00b56a29-e237-4a33-bd1a-c4ee43090121.png)

**Comment by chrisron95** [comment]
`echo` is defined at the top of the automation in the `variables` part of the sequence. I'm not sure how google home works as I only have Amazon Echos currently, so I'm not sure how their service calls work but I'm sure you could adapt it to work if you adjust the room names to correlate to the names of your google home devices. Then you're going to have to adjust the service calls in the rest of the script. I don't believe the tags used in the message will work for google home so you're going to have to do your own research on how adjust the voices, if that can be done. Feel free to post your yaml code of your automation if you want and I may be able to help you tweak it to work.

**Comment by CaoHoa1** [comment]
Thanks for the reply this is all my code





group:
  googles:
    name: All Googles
    entities:
      - media_player.minhtue
      - media_player.minhthong_room
      - media_player.van_phong
      - media_player.van_phong_1
      - media_player.master_bedroom_display
      - media_player.bedroom_hub
      - media_player.livingroomcc
     
  googles_main:
    name: All Main Googles
    entities:
      - media_player.van_phong
      - media_player.minhtue  
      - media_player.van_phong
      - media_player.van_phong_1
      - media_player.bedroom_hub
      - media_player.livingroomcc
      
     
mqtt:
  sensor:
    - name: "Jarvis Last Msg"
      state_topic: "house/speech/lastmsg"
    - name: "Jarvis Last Location"
      state_topic: "house/speech/lastloc"
    - name: "Jarvis Last Msg Time"
      state_topic: "house/speech/msgtime"

template:
  - sensor:
      - name: last_google
        state: >
          {{ expand('group.googles') | selectattr('attributes.last_called','eq',True) | map(attribute='entity_id') | first }}
        availability: >
          {{ expand('group.googles') | selectattr('attributes.last_called','eq',True) | first is defined }}
 

input_select:
  google_voice:
    name: Giọng nói
    options:
      - Red
      - Orange
      - Amber
    initial: Red


script:
  
  ###############################################################################
  # Voice Notify
  # Conditions:
  #   => Only Announce when people are home.
  #   => Only Announce when Audible Notifications are on
  ###############################################################################

  speech_engine:
    alias: giọng nói trợ lý 
    icon: "mdi:account-voice"
    mode: 'parallel'
    sequence:
      - condition: state
        entity_id: input_boolean.audible_notifications
        state: "on"
      #  - condition: state
      #    entity_id: group.family
      #    state: "home"
      - condition: state
        entity_id: input_boolean.vacation_mode
        state: "off"
      - service: script.google_voice
        data:
          who: "{{ who }}"
          message: >
            {{ message }}

  # Called wheh notification needs to play on echos
  google_voice:
    alias: giọng nói trợ lý alexa
    icon: "mdi:account-voice"
    mode: 'parallel'
    sequence:
      - variables:
          goole: >
            {% set loc = states('sensor.room_presence') if who in ['presence','room_presence','current','current_room'] else who %}
            {% if loc in ['van_phong','media_player.van_phong','office'] %}
              van_phong
            {% elif loc in ['van_phong_1','media_player.van_phong_1','office'] %}
              van_phong_1
            {% elif loc in ['minhtue','media_player.minhtue','living_room','dining_room'] %}
              minhtue
            {% elif loc in ['minhthong_room','media_player.minhthong_room','lab'] %}
              minhthong_room
            {% elif loc in ['livingroomcc','media_player.livingroomcc','chromecast_audio','hallway'] %}
              livingroomcc
            {% elif loc in ['bedroom hub','media_player.bedroom_hub','fire','chromecast_audio','bedroom_hub'] %}
              bedroom hub  
            {% elif loc in ['google_assistant','media_player.google_assistant','garage'] %}
              google_assistant
            {% elif loc in ['everywhere'] %}
              everywhere
            {% else %}
              livingroomcc
            {% endif %}
          volume: >
            {%- set default = '0.6' if states('input_boolean.audible_notifications') == 'on' else '0.3' -%}
            {%- if volume_override is float -%}
              {{ volume_override }}
            {%- else -%}
              {{ default }}
            {%- endif -%}
      - alias: Save your message for recall
        service: mqtt.publish
        data_template:
          topic: "house/speech/lastmsg"
          payload: >
            {% if msg_summary == 'recall' %}
              {{- states('sensor.jarvis_last_msg') -}}
            {% else %}
              {% if msg_summary %}
                {% set message = msg_summary %}
              {% endif %}
              {%- macro cleanup(data) -%}
                {%- for item in data.split("\n")  if item | trim != "" -%}
                  {{ item | trim }} {% endfor -%}
              {%- endmacro -%}
              {{- cleanup( message | striptags | truncate(220) ) -}}
            {% endif %}
      - alias: Save time of last message
        service: mqtt.publish
        data_template:
          topic: "house/speech/msgtime"
          payload: >
            {{ now().strftime("%-I") }}:{{ now().strftime("%M") }} {{ now().strftime("%p") }}
          retain: true
      - alias: Save location
        service: mqtt.publish
        data_template:
          topic: "house/speech/lastloc"
          payload: "{{ tts_entity }}"
          retain: true
      - alias: Set google volume if needed
        if:
          - condition: not
            conditions:
              - condition: template
                value_template: >-
                  {{ is_state_attr('media_player.' + tts_entity, 'volume_level', volume | float) }}
        then:
          - service: media_player.volume_set
            data:
              volume_level: "{{ volume | float }}"
            target:
              entity_id: media_player.{{ tts_entity }}
      - alias: Send the notification
        service: tts.google_say
        data:
          message: >
            <voice name="{{ states('input_select.google_voice') }}">
            <google:auto-breaths>
            <prosody rate="fast">
            {{ message }}
            </prosody>
            </google:auto-breaths>
            </voice>
          target: >
            {% if tts_entity == 'everywhere' %}
              group.googles_main
            {% else %}
              media_player.{{ tts_entity }}
            {% endif %}
          data:
            type: tts

**Comment by CaoHoa1** [comment]
Do you use UltraViewer? Can you go to my computer and edit it for me. This above does not support sending yaml files
